tests/trx_generator/http_client_async.hpp

```diff
@@ -0,0 +1,166 @@
+#pragma once
+
+// The majority of the code here are derived from boost source
+// libs/beast/example/http/client/async/http_client_async.cpp
+// with minimum modification and yet reusable.
+//
+//
+// Copyright (c) 2016-2019 Vinnie Falco (vinnie dot falco at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+//------------------------------------------------------------------------------
+//
+// Example: HTTP client, asynchronous
+//
+//------------------------------------------------------------------------------
+
+#include <boost/asio/strand.hpp>
+#include <boost/beast/core.hpp>
+#include <boost/beast/http.hpp>
+#include <boost/beast/version.hpp>
+#include <cstdlib>
+#include <functional>
+#include <iostream>
+#include <memory>
+#include <string>
+
+namespace beast = boost::beast;         // from <boost/beast.hpp>
+namespace http  = beast::http;          // from <boost/beast/http.hpp>
+namespace net   = boost::asio;          // from <boost/asio.hpp>
+using tcp       = boost::asio::ip::tcp; // from <boost/asio/ip/tcp.hpp>
+
+//------------------------------------------------------------------------------
+
+namespace eosio {
+
+using response_callback_t = std::function<void(beast::error_code ec, http::response<http::string_body>)>;
```

Comment:
Seems like another namespace of `http_client_async` would be good for this and the function at the end.

Reply:
Addressed: [f02287e](https://github.com/AntelopeIO/leap/pull/1052/commits/f02287e44535e51b7ae21d1c90db73c40f51d644)